sdk/platform/http_client/curl/src/curl_client.cpp

```diff
@@ -32,5 +32,117 @@ Response CurlClient::send()
     }
   }
 
-  return Response(200, "OK\n");
+  return this->m_response;
+}
+
+CURLcode CurlClient::Perform()
+{
+  m_firstHeader = true;
+
+  auto settingUp = SetUrl();
+  if (settingUp != CURLE_OK)
+  {
+    return settingUp;
+  }
+
+  settingUp = SetHeaders();
+  if (settingUp != CURLE_OK)
+  {
+    return settingUp;
+  }
+
+  settingUp = SetWriteResponse();
+  if (settingUp != CURLE_OK)
+  {
+    return settingUp;
+  }
+
+  return curl_easy_perform(m_pCurl);
+}
+
+static void ParseAndSetFirstHeader(std::string const& header, Response** response)
```

Comment:
Hasn't curl already done this parsing for us?

Reply:
curl parses status code only.  I need to get minor and mayor version. And also reason phrase.
Also, we need to wait until perform() is completed to read those values from curl.  But we need to create the response as soon as we get the first header

Reply:
If curl doesn't parse those it suggests that customers have not needed that information. Why does the rest of the SDK care about which specific HTTP transport is in use?

Reply:
We absolutely need the status code. We probably don't need the version info. The reason phase could be useful for logging (if nothing else).